core/src/main/java/forklift/consumer/Consumer.java

```diff
@@ -292,20 +297,72 @@ public void inject(ForkliftMessage msg, final Object instance) {
                                     }
                                 } catch (Exception e) {
                                     log.debug("", e);
-                                }                             
+                                }
                             }
                         } else if (decorator == Config.class) {
+                            final forklift.decorators.Config annotation = f.getAnnotation(forklift.decorators.Config.class);
                             if (clazz == Properties.class) {
-                                forklift.decorators.Config config = f.getAnnotation(forklift.decorators.Config.class);
-                                f.set(instance, PropertiesManager.get(config.value()));
+                                String confName = annotation.value();
+                                if (confName.equals("")) {
+                                    confName = f.getName();
+                                }
+                                final Properties config = PropertiesManager.get(confName);
+                                if (config == null) {
+                                    log.warn("Attempt to inject field failed because resource file {} was not found", annotation.value());
+                                    return;
+                                }
+                                f.set(instance, config);
+                            } else {
+                                final Properties config = PropertiesManager.get(annotation.value());
+                                if (config == null) {
+                                    log.warn("Attempt to inject field failed because resource file {} was not found", annotation.value());
+                                    return;
+                                }
+                                String key = annotation.field();
+                                if (key.equals("")) {
+                                    key = f.getName();
+                                }
+                                Object value = config.get(key);
+                                if (value != null) {
```

Comment:
Is there a reason to check for a null value rather than just setting the field? Eliminates a little code for a bad wired dependency.